source/binary.cpp

```diff
@@ -136,13 +137,17 @@ class Parser {
   // On failure, returns an error code and issues a diagnostic.
   spv_result_t parseInstruction();
 
-  // Parses an instruction operand with the given type.
-  // May update the expected_operands parameter, and the scalar members of the
-  // inst parameter. On success, returns SPV_SUCCESS, advances past the
-  // operand, and pushes a new entry on to the operands vector.  Otherwise
-  // returns an error code and issues a diagnostic.
-  spv_result_t parseOperand(spv_parsed_instruction_t* inst,
+  // Parses an instruction operand with the given type, for an instruction
+  // starting at inst_offset words into the SPIR-V binary.  Appends, to the
+  // words parameter, the endian-translated words for the operand, but only if
```

Comment:
The tipping point on the complexity of this is caused by the fact that the method behaves differently based on whether endian conversion must be done or not.  If no endian conversion is required, then the endian_converted_inst_words parameter is ignored.  Life would be simpler if we didn't mind copying the underlying data in that case.  We save on allocations and copying time in that very common case.  I think it's worth keeping that optimization.

Maybe I worded it poorly.  How about replacing the sentence from "Appends....from host" with:

If the SPIR-V binary is the same endianness as the host, then the endian_converted_inst_words parameter is ignored.  Otherwise, this method appends the words for this operand, converted to host native endianness, to the end of endian_converted_inst_words.